File: HTML_CSS_JS/src/fishing/fishing.js
// fishing.js - connector that mirrors the original Lua `fishing.lua`
(function(){
  const Fishing = {
    init(){ if(window.FishingController && window.FishingController.init) window.FishingController.init() },
    update(dt){
      // pass through weather/daynight if available
      const Weather = window.Weather || null
      const DayNight = window.DayNightCycle || null
      const weatherData = Weather && Weather.getWeatherData ? Weather.getWeatherData() : null
      if(window.FishingController && window.FishingController.update) window.FishingController.update(dt, weatherData, DayNight)
    },
    draw(ctx){
      // if controller not idle, draw rod and call CastingSystem.draw
      try{
        const FC = window.FishingController
        if(!FC) return
        if(FC.getState && FC.getState() !== 'idle'){
          const player = window.Player || { x: 400, y: 300, w:16, h:32 }
          const playerX = player.x || 400
          const playerY = player.y || 300
          const playerW = player.w || 16
          const playerH = player.h || 32

          // draw rod: simple line from player's hand
          const rodStartX = playerX + playerW/3
          const rodStartY = playerY - playerH/4
          const rodEndX = playerX + 40
          const rodEndY = playerY - playerH - 30
          if(ctx){
            ctx.save(); ctx.strokeStyle = 'rgba(100,60,30,1)'; ctx.lineWidth = 4; ctx.beginPath(); ctx.moveTo(rodStartX, rodStartY); ctx.lineTo(rodEndX, rodEndY); ctx.stroke();
            // handle
            ctx.strokeStyle = 'rgba(70,50,30,1)'; ctx.lineWidth = 6; const handleEndX = rodStartX + (rodEndX-rodStartX)*0.25; const handleEndY = rodStartY + (rodEndY-rodStartY)*0.25; ctx.beginPath(); ctx.moveTo(rodStartX,rodStartY); ctx.lineTo(handleEndX,handleEndY); ctx.stroke(); ctx.restore();
          }

          // call casting draw
          if(window.CastingSystem && window.CastingSystem.draw) window.CastingSystem.draw(ctx, rodEndX, rodEndY, FC.getState(), FC.isBiting && FC.isBiting())
        }
      }catch(e){ console.warn('Fishing.draw error', e) }
    },
  // Delegates / helpers
    getState(){ return (window.FishingController && window.FishingController.getState) ? window.FishingController.getState() : 'idle' },
    setState(s){ if(window.FishingController && window.FishingController.setState) window.FishingController.setState(s) },
    getBaitInventory(){ return (window.FishingController && window.FishingController.getBaitInventory) ? window.FishingController.getBaitInventory() : [] },
    getCurrentBait(){ return (window.FishingController && window.FishingController.getCurrentBait) ? window.FishingController.getCurrentBait() : null },
    selectBait(i){ if(window.FishingController && window.FishingController.selectBait) window.FishingController.selectBait(i) },
    toggleCasting(){ if(window.FishingController && window.FishingController.toggleCasting) return window.FishingController.toggleCasting(); return false },
    attemptHookSet(){ if(window.FishingController && window.FishingController.attemptHookSet) return window.FishingController.attemptHookSet(); return false },
    manualReel(){ if(window.FishingController && window.FishingController.manualReel) return window.FishingController.manualReel(); return false },
    forceBite(){ if(window.FishingController && window.FishingController.forceBite) return window.FishingController.forceBite() },
    getCastPower(){ return (window.CastingSystem && window.CastingSystem.getCastPower) ? window.CastingSystem.getCastPower() : 0 },
    getTotalCatches(){ return (window.FishingController && window.FishingController.getTotalCatches) ? window.FishingController.getTotalCatches() : 0 },
    getNightsCaught(){ return (window.FishingController && window.FishingController.getNightsCaught) ? window.FishingController.getNightsCaught() : 0 },
    getLineStress(){ return (window.FishingController && window.FishingController.getLineStress) ? window.FishingController.getLineStress() : 0 },
    getMaxLineStress(){ return (window.FishingController && window.FishingController.getMaxLineStress) ? window.FishingController.getMaxLineStress() : 1 },
    isFishPulling(){ return (window.FishingController && window.FishingController.isFishPulling) ? window.FishingController.isFishPulling() : false },
    getReelProgress(){ return (window.FishingController && window.FishingController.getReelProgress) ? window.FishingController.getReelProgress() : 0 },
    getMaxReelProgress(){ return (window.FishingController && window.FishingController.getMaxReelProgress) ? window.FishingController.getMaxReelProgress() : 1 },
    getStrugglingIntensity(){ return (window.FishingController && window.FishingController.getStrugglingIntensity) ? window.FishingController.getStrugglingIntensity() : 0 },
    getCurrentCatch(){ return (window.FishingController && window.FishingController.getCurrentCatch) ? window.FishingController.getCurrentCatch() : null },
    isBiting(){ return (window.FishingController && window.FishingController.isBiting) ? window.FishingController.isBiting() : false },
    getBiteTimer(){ return (window.FishingController && window.FishingController.getBiteTimer) ? window.FishingController.getBiteTimer() : 0 },
    getBiteDelay(){ return (window.FishingController && window.FishingController.getBiteDelay) ? window.FishingController.getBiteDelay() : 0 },
    keypressed(k){ if(window.FishingController && window.FishingController.keypressed) window.FishingController.keypressed(k) }
  }

  // expose to global for file:// usage
  if(typeof window !== 'undefined') window.Fishing = window.Fishing || Fishing
  if(typeof module !== 'undefined') module.exports = Fishing
})();

File: Love2d_Web/fishing/fishing.lua
-- fishing.lua - Fishing System Connector - connects all fishing components
local Fishing = {}

-- Import the main fishing controller (which handles all subsystems)
local FishingController = require("fishing.fishingcontroller")
local CastingSystem = require("fishing.castingsystem")

-- Initialize all fishing components (delegate to controller)
function Fishing:init()
    FishingController:init()
end

-- Update all fishing components (delegate to controller)
function Fishing:update(dt)
    local Weather = require("world.weather")
    local DayNightCycle = require("world.daynightcycle")
    local weatherData = Weather:getWeatherData()
    FishingController:update(dt, weatherData, DayNightCycle)
end

-- Draw all fishing components
function Fishing:draw()
    if FishingController:getState() ~= "idle" then
        local playerX, playerY = require("player"):getPosition()
        local playerWidth, playerHeight = require("player"):getDimensions()
        
        -- Draw fishing rod (more realistic angle and position)
        love.graphics.setColor(0.4, 0.25, 0.1, 1) -- Darker brown for rod
        love.graphics.setLineWidth(4)
        
        -- Rod extends from player's right hand position at an angle
        local rodStartX = playerX + playerWidth/3  -- Start more to the right
        local rodStartY = playerY - playerHeight/4 -- Start from player's hand level
        
        -- Add rod bend effect when fish is biting or struggling
        local fishingState = FishingController:getState()
        local isBiting = self:isBiting()
        local bendOffset = 0
        
        if isBiting or fishingState == "struggling" or fishingState == "reeling" then
            -- Rod bends under pressure
            local time = love.timer.getTime()
            bendOffset = math.sin(time * 8) * 3 + 5 -- Rod bends down
        end
        
        local rodEndX = playerX + 40  -- Extend further right
        local rodEndY = playerY - playerHeight - 30 + bendOffset  -- Higher up, with bend
        
        -- Draw the rod
        love.graphics.line(rodStartX, rodStartY, rodEndX, rodEndY)
        
        -- Draw rod handle (thicker part near player)
        love.graphics.setColor(0.3, 0.2, 0.1, 1)
        love.graphics.setLineWidth(6)
        local handleLength = 0.25
        local handleEndX = rodStartX + (rodEndX - rodStartX) * handleLength
        local handleEndY = rodStartY + (rodEndY - rodStartY) * handleLength
        love.graphics.line(rodStartX, rodStartY, handleEndX, handleEndY)
        
        -- Reset line width and draw the casting system
        love.graphics.setLineWidth(1)
        local isBiting = self:isBiting()
        CastingSystem:draw(rodEndX, rodEndY, FishingController:getState(), isBiting)
    end
end

function Fishing:drawUI()
    -- UI drawing is handled by the original systems in main.lua
    -- Individual components don't have separate drawUI methods
end

-- Delegate all methods to FishingController
function Fishing:getState()
    return FishingController:getState()
end

function Fishing:setState(state)
    return FishingController:setState(state)
end

function Fishing:getBaitInventory()
    return FishingController:getBaitInventory()
end

function Fishing:getCurrentBait()
    return FishingController:getCurrentBait()
end

function Fishing:selectBait(index)
    return FishingController:selectBait(index)
end

function Fishing:toggleCasting()
    return FishingController:toggleCasting()
end

function Fishing:attemptHookSet()
    return FishingController:attemptHookSet()
end

function Fishing:manualReel()
    return FishingController:manualReel()
end

function Fishing:forceBite()
    return FishingController:forceBite()
end

function Fishing:getCastPower()
    return FishingController:getCastPower()
end

function Fishing:getTotalCatches()
    return FishingController:getTotalCatches()
end

function Fishing:getNightsCaught()
    return FishingController:getNightsCaught()
end

function Fishing:getLineStress()
    return FishingController:getLineStress()
end

function Fishing:getMaxLineStress()
    return FishingController:getMaxLineStress()
end

function Fishing:isFishPulling()
    return FishingController:isFishPulling()
end

function Fishing:getReelProgress()
    return FishingController:getReelProgress()
end

function Fishing:getMaxReelProgress()
    return FishingController:getMaxReelProgress()
end

function Fishing:getStrugglingIntensity()
    return FishingController:getStrugglingIntensity()
end

function Fishing:getCurrentCatch()
    return FishingController:getCurrentCatch()
end

function Fishing:isBiting()
    return FishingController:isBiting()
end

function Fishing:getBiteTimer()
    return FishingController:getBiteTimer()
end

function Fishing:getBiteDelay()
    return FishingController:getBiteDelay()
end

function Fishing:keypressed(key)
    return FishingController:keypressed(key)
end

return Fishing
